### DISCUSSION TITLE
kafka输出格式问题咨询

### DISCUSSION THREAD

**Discussion by XTUxiongda** [alibaba/ilogtail#1083] [Help] [answered]
`enable: true
inputs:
  - Type: file_log
    LogPath: /logs/fofund-ipmc/
    FilePattern: "*.log"
processors:
  - Type: processor_json
    SourceKey: content
    KeepSource: false
    ExpandDepth: 1
    ExpandConnector: ""
flushers:
  - Type: flusher_stdout
    OnlyStdout: true
  - Type: flusher_kafka
    Brokers: 
      - 10.105.100.62:9092
    Topic: ipmc-log`

上述是我的配置文件。
我现在遇到的问题是，明明控制台的输出是一串json，但是写到kafka的数据会变成拼接了key和Value的数组，如下
`{
                "Time":1692756272,
                "Contents":[
                    {
                        "Key":"__tag__:__path__",
                        "Value":"/logs/fofund-ipmc/fofund-ipmc.2023-08-23.error.0.log"
                    },
                    {
                        "Key":"__tag__:__user_defined_id__",
                        "Value":"default"
                    },
                    {
                        "Key":"appName",
                        "Value":"fofund-ipmc"
                    },
                    {
                        "Key":"serverIP",
                        "Value":"10.11.63.40:8080"
                    },
                    {
                        "Key":"traceId",
                        "Value":"N/A"
                    },
                    {
                        "Key":"createTime",
                        "Value":"2023-08-23 10:04:32 158"
                    },
                    {
                        "Key":"level",
                        "Value":"ERROR"
                    },
                    {
                        "Key":"thread",
                        "Value":"main"
                    },
                    {
                        "Key":"message",
                        "Value":"获取id服务的id时异常，id取值逻辑保持原样"
                    }
                ],
                "Time_ns":0
            }`

我希望Contents里面都变成一串json和控制台输出一样，没有拼接Key,Value这一堆是可以实现的吗

**Comment by 7y-9** [comment]
把processor_json去掉试一下呢

**Reply by XTUxiongda** [reply]
还是一样

**Reply by XTUxiongda** [reply]
@shalousun  您好，这个问题和您说的后期加一个协议打平的是同一个问题么

**Comment by XTUxiongda** [comment]
我使用的ilogtail的版本是1.7.0

**Reply by messixukejia** [reply]
flusher使用flusher_kafka_v2 https://ilogtail.gitbook.io/ilogtail-docs/data-pipeline/flusher/flusher-kafka_v2#yang-li

**Reply by XTUxiongda** [reply, marked as answer]
<img width="1174" alt="08f655826b1353838144f66df8f5b66" src="https://github.com/alibaba/ilogtail/assets/38322373/d55ad0f0-5b99-4458-8bef-930424c73a67">
kafka_v2中的样例描述有误， Type应写成flusher_kafka_v2

**Comment by XTUxiongda** [comment]
> flusher使用flusher_kafka_v2 https://ilogtail.gitbook.io/ilogtail-docs/data-pipeline/flusher/flusher-kafka_v2#yang-li
